### PR TITLE
Move numeric consts to associated consts

### PR DESCRIPTION
**Pull request by faern**
Implements RFC https://github.com/rust-lang/rfcs/pull/2700 (which at the time of writing this is not yet merged).

Since the RFC has `disposition-merge` and the discussion does not voice any opinions against it, I don't have any reason to believe it won't be merged. And I felt that showing the code needed to do this would give the RFC a push again.

One thing that is not yet implemented is where the compiler prints `std::i8::MIN` and similar in error/help messages. These now need to be `i8::MIN` instead, but I don't know where the compiler get this name from.

It still also does not do the required changes to the `src/stdarch` submodule. I did not want to get in too deep before this has been discussed/accepted.

### CODE REVIEW

File: src/libcore/num/f32.rs [hunk 1]
```diff
@@ -153,6 +167,60 @@ pub mod consts {
...
+    /// [Machine epsilon]: https://en.wikipedia.org/wiki/Machine_epsilon
+    #[stable(feature = "assoc_int_consts", since = "1.42.0")]
+    pub const EPSILON: f32 = 1.19209290e-07_f32;
+
+    /// Smallest finite `f32` value.
+    #[stable(feature = "assoc_int_consts", since = "1.42.0")]
+    pub const MIN: f32 = -3.40282347e+38_f32;
+    /// Smallest positive normal `f32` value.
+    #[stable(feature = "assoc_int_consts", since = "1.42.0")]
+    pub const MIN_POSITIVE: f32 = 1.17549435e-38_f32;
```
Comment (Centril) [inline src/libcore/num/f32.rs]: The non-associated constants should use the definition from the associated ones to have one source of truth.

Comment (faern) [reply src/libcore/num/f32.rs]: Done

File: src/libstd/collections/hash/map.rs [hunk 2]
```diff
@@ -2619,7 +2619,7 @@ mod test_map {
     use crate::cell::RefCell;
     use rand::{thread_rng, Rng};
     use realstd::collections::TryReserveError::*;
-    use realstd::usize;
+    use realusize;
```
[the remark this answers is not in the document]
Comment (faern) [reply src/libstd/collections/hash/map.rs]: Oops. Yeah some of this is just search-and-replace fixed. I was hoping `./x.py build` would catch them. But apparently not all of them.

File: src/libstd/io/cursor.rs [hunk 3]
```diff
@@ -939,7 +939,7 @@ mod tests {
     #[cfg(target_pointer_width = "32")]
     fn vec_seek_and_write_past_usize_max() {
         let mut c = Cursor::new(Vec::new());
-        c.set_position(<usize>::max_value() as u64 + 1);
+        c.set_position(<usize>::MAX as u64 + 1);
```
[the remark this answers is not in the document]
Comment (faern) [reply src/libstd/io/cursor.rs]: Sure. I just opted to change as little as possible myself. I don't want the RFC/PR to get stalled on something that is irrelevant to the actual moving of the constants. But this seems small enough.

**Comment by JohnCSimon** [comment]
Ping from triage: 
@faern - Can you please address the merge conflicts and build failures? Thanks.

**Comment by faern** [comment]
@JohnCSimon I was under the impression that we did not want to merge all the deprecation and everything in one go. But rather first introduce the new constants only and deprecate them later. See discussion in https://github.com/rust-lang/rfcs/pull/2700 and heavily reduced PR at #68325.

That PR has a lot less risk of experiencing bit rot, since it does not touch so much of the standard library/compiler.

**Comment by Dylan-DPC** [comment]
@faern since you are splitting this PR into multiple ones, is it fine if we close this? Thanks

**Comment by faern** [comment]
> @faern since you are splitting this PR into multiple ones, is it fine if we close this? Thanks

My intention was to use this PR to update the entire repo to use the new constants (a subset of what this PR currently does) after they have been stabilized (hopefully soon). But if you prefer me to open a new PR for that instead of keeping this one open you are free to close it.

**Comment by Dylan-DPC** [comment]
I think it is better idea to close it. Thanks

**Pull request closed without merging**